File: backend/app/schemas/reconciliation.py
import uuid
from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel


class ReconciliationRunRequest(BaseModel):
    location_id: uuid.UUID | None = None
    period_start: datetime
    period_end: datetime


class ReconciliationFlagResponse(BaseModel):
    model_config = {"from_attributes": True}

    id: uuid.UUID
    run_id: uuid.UUID
    flag_type: str
    severity: str
    message: str
    document_id: uuid.UUID | None
    expense_id: uuid.UUID | None
    toast_order_id: uuid.UUID | None
    is_resolved: bool
    resolved_by: uuid.UUID | None
    resolved_at: datetime | None
    resolution_note: str | None
    created_at: datetime


class ReconciliationRunResponse(BaseModel):
    model_config = {"from_attributes": True}

    id: uuid.UUID
    tenant_id: uuid.UUID
    location_id: uuid.UUID | None
    period_start: datetime
    period_end: datetime
    status: str
    triggered_by: uuid.UUID | None
    started_at: datetime | None
    completed_at: datetime | None
    error_message: str | None
    documents_checked: int
    expenses_checked: int
    toast_orders_checked: int
    flags_raised: int
    total_sales_amount: Decimal | None
    total_expense_amount: Decimal | None
    net_variance: Decimal | None
    created_at: datetime


class ResolveFlagRequest(BaseModel):
    resolution_note: str

File: frontend/src/types/reconciliation.ts
export type ReconciliationStatus = "pending" | "running" | "complete" | "failed";
export type FlagSeverity = "low" | "medium" | "high" | "critical";
export type FlagType =
  | "missing_invoice"
  | "duplicate_invoice"
  | "duplicate_expense"
  | "uncategorized_expense"
  | "suspicious_amount"
  | "unmatched_sale"
  | "unverified_payroll";

export interface ReconciliationRun {
  id: string;
  tenant_id: string;
  location_id: string | null;
  period_start: string;
  period_end: string;
  status: ReconciliationStatus;
  triggered_by: string | null;
  started_at: string | null;
  completed_at: string | null;
  error_message: string | null;
  documents_checked: number;
  expenses_checked: number;
  toast_orders_checked: number;
  flags_raised: number;
  total_sales_amount: string | null;
  total_expense_amount: string | null;
  net_variance: string | null;
  created_at: string;
}

export interface ReconciliationFlag {
  id: string;
  run_id: string;
  flag_type: FlagType;
  severity: FlagSeverity;
  message: string;
  document_id: string | null;
  expense_id: string | null;
  toast_order_id: string | null;
  is_resolved: boolean;
  resolved_by: string | null;
  resolved_at: string | null;
  resolution_note: string | null;
  created_at: string;
}

export interface RunListMeta {
  page: number;
  limit: number;
  total: number;
}
